servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -978,6 +983,111 @@ boolean shouldRetryOperation(ResourceException ex, int retryCount) {
+
+                // first verify that auditing requirements are met
+
+                checkDomainAuditEnabled(con, domainName, auditRef, caller, getPrincipalName(ctx), AUDIT_TYPE_POLICY);
+
+                // verify the policy hasn't reached maximum number of versions allowed
+
+                List<String> policyVersions = con.listPolicyVersions(domainName, policyName);
+                if (policyVersions.size() >= maxPolicyVersions) {
+                    con.rollbackChanges();
+                    throw ZMSUtils.quotaLimitError("unable to put policy: " + policyName + ", version: " + version + ", max number of versions reached (" + maxPolicyVersions + ")", caller);
+                }
+
+                // retrieve our source policy version
+
+                Policy originalPolicy = getPolicy(con, domainName, policyName, fromVersion);
+
+                // check that quota is not exceeded
+
+                quotaCheck.checkPolicyQuota(con, domainName, originalPolicy, caller);
+
+                // now process the request
+
+                StringBuilder auditDetails = new StringBuilder(ZMSConsts.STRING_BLDR_SIZE_DEFAULT);
+                originalPolicy.setVersion(version);
+                if (!con.insertPolicy(domainName, originalPolicy)) {
+                    con.rollbackChanges();
+                    throw ZMSUtils.internalServerError("unable to put policy: " + originalPolicy.getName() + ", version: " + version, caller);
+                }
+
+                // open our audit record
+
+                auditDetails.append("{\"name\": \"").append(policyName);
+                auditDetails.append("\", \"version\": \"").append(version);
+
+                // now we need process our policy assertions
+
+                List<Assertion> newAssertions = originalPolicy.getAssertions();
+                if (newAssertions != null) {
+                    for (Assertion assertion : newAssertions) {
+
+                        // get assertion conditions for original assertion
+
+                        AssertionConditions assertionConditions = new AssertionConditions();
+                        if (assertion.getId() != null) {
+                            assertionConditions.setConditionsList(con.getAssertionConditions(assertion.getId()));
+                        }
+
+                        // insert assertion (and get new assertion id)
+
+                        if (!con.insertAssertion(domainName, policyName, version, assertion)) {
+                            con.rollbackChanges();
+                            throw ZMSUtils.internalServerError("unable to put policy: " + originalPolicy.getName() + ", version: " + version + ", fail inserting assertion", caller);
+                        }
+
+                        // copy assertion conditions for new assertion id
+
+                        if (assertionConditions != null && assertionConditions.getConditionsList() != null && !assertionConditions.getConditionsList().isEmpty()) {
```

Comment:
on line 1038 we have: AssertionConditions assertionConditions = new AssertionConditions(); 

so there is no need to call assertionConditions != null since it will never be null

Reply:
fixed